DisposableAnnotations/DisposesAttribute.cs

```diff
@@ -0,0 +1,12 @@
+namespace DisposableAnnotations
+{
+    using System;
+
+    /// <summary>
+    /// The containing method owns the instance and is responsible for disposing it.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = true)]
+    public class DisposesAttribute : Attribute
```

Comment:
To come up with an alternative for the sake of discussion: `[BecomesOwner]`? `[TransferOwnership]`?

`[Disposes]` sounds like the method itself does the disposing.